pkg/kubelet/cm/devicemanager/plugin/v1beta1/server.go

```diff
@@ -103,25 +174,84 @@ func (s *server) Start() error {
 		return err
 	}
 
-	ln, err := net.Listen("unix", s.SocketPath())
+	api.RegisterRegistrationServer(s.grpc, s)
+	s.ctx, s.cancel = context.WithCancel(context.Background())
+	s.wg.Add(1)
+	go func(ctx context.Context) {
+		defer s.wg.Done()
+		s.serveWithRetry(ctx)
+	}(s.ctx)
+
+	return nil
+}
+
+// serveWithRetry serves the gRPC server with retry.
+func (s *server) serveWithRetry(ctx context.Context) {
+	for {
+		s.setHealthy()
+		serveErrCh := make(chan error, 1)
+		s.wg.Add(1)
+		go func() {
+			defer s.wg.Done()
+			err := s.serve()
+			serveErrCh <- err
+			close(serveErrCh)
+			s.setUnhealthy()
+		}()
```

Comment:
Just wondering if it's really needed to use both ways of blocking: by the working group and by the context? Would using only context be enough? `serveErrCh` can be also replaced by the [context.WithCancelCause](https://pkg.go.dev/context#WithCancelCause) it seems.